framework/db/ExpressionBuilderTrait.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace yii\db;
+
+/**
+ * Trait ExpressionBuilderTrait provides common constructor for classes that
+ * should implement [[ExpressionBuilderInterface]]
+ *
+ * @author Dmytro Naumenko <d.naumenko.a@gmail.com>
+ * @since 2.0.14
+ */
+trait ExpressionBuilderTrait
```

Comment:
i see no usecase where you need splitted interface and trait, i would merge ExpressionBuilderTrait and ExpressionBuilderInterface into abstract class

Reply:
Disagreed. I can build my own extending tree and implement class wherever I want.
Abstract class forces me to add it in extending tree root and makes reusing harder.

Reply:
you're right, separation will be needed for example in https://github.com/yiisoft/yii2/pull/15348/files#r156621010